src/Chapter2.hs

```diff
@@ -841,7 +857,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate c l
+ | c < 0 = []
+ | length l == 0 = []
```

Comment:
That is a correct idea!

However, as I mentioned above, `length` is not cheap for lists in Haskell 😞 
But fortunately, there is a way to implement the same check, but for free. You can just compare the list to an empty list. It will use pattern match and will work instantly. For that, there is a standard `null` function as well:
```suggestion
 | null l = []
```

Reply:
Yes, will keep the cost of using length function in mind.